src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc
             using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))
             {
                 var buffer = CreateTextBuffer(reader, cancellationToken);
-                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);
+                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);
```

Comment:
In the case of no encoding this will end up putting a BOM into the file in question.  Is that intentional?  My expectation is that we would insert no BOM as this is how most of the world works.